editor/editor_properties.cpp

```diff
@@ -2183,7 +2183,55 @@ void EditorPropertyResource::_menu_option(int p_which) {
 		case OBJ_MENU_NEW_SCRIPT: {
 
 			if (Object::cast_to<Node>(get_edited_object())) {
-				EditorNode::get_singleton()->get_scene_tree_dock()->open_script_dialog(Object::cast_to<Node>(get_edited_object()));
+				if (get_edited_property() == "script") {
+					EditorNode::get_singleton()->get_scene_tree_dock()->open_script_dialog(Object::cast_to<Node>(get_edited_object()));
+				} else {
+					// Copied from editor/scene_tree_dock.cpp#L380 and edited for our purposes
+					// There's a good chance we could merge this back into that code with some modifications
+					//Ref<Script> existing = get_edited_object()->get(get_edited_property());
+					//Script *existing = Object::cast_to<Script>(get_edited_object()->get(get_edited_property()));
```

Comment:
No need to keep commented out code.
`Ref<Script>` wasn't working? It seems more explicit than `RES`, and would make sure that it's not considered as valid if it's not a Script resource.

Reply:
I probably should try `Ref<Script>` again, as I was having to deal with a crash when it was calling `get_class` before calling `is_valid` (which was weird to me since scene_tree_dock.cpp did it without any issues), and only realized I was trying to call a method on a non-existant object after having switched over to `RES`.